ee/server/service/mdm.go

```diff
@@ -285,3 +289,121 @@ func (svc *Service) DeleteMDMAppleSetupAssistant(ctx context.Context, teamID *ui
+
+	logging.WithLevel(logging.WithNoUser(ctx), level.Info)
+
+	appConfig, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return "", ctxerr.Wrap(ctx, err, "getting app config")
+	}
+
+	settings := appConfig.MDM.EndUserAuthentication.SSOProviderSettings
+
+	// For now, until we get to #10999, we assume that SSO is disabled if
+	// no settings are provided.
+	if settings.IsEmpty() {
+		err := &fleet.BadRequestError{Message: "organization not configured to use sso"}
+		return "", ctxerr.Wrap(ctx, err, "initiate mdm sso")
+	}
+
+	metadata, err := sso.GetMetadata(&settings)
+	if err != nil {
+		return "", ctxerr.Wrap(ctx, err, "InitiateSSO getting metadata")
+	}
+
+	serverURL := appConfig.ServerSettings.ServerURL
+	authSettings := sso.Settings{
+		Metadata:                    metadata,
+		AssertionConsumerServiceURL: serverURL + svc.config.Server.URLPrefix + "/mdm/apple/sso",
+		SessionStore:                svc.ssoSessionStore,
+		OriginalURL:                 "/api/v1/fleet/mdm/sso/callback",
+	}
+
+	idpURL, err := sso.CreateAuthorizationRequest(&authSettings, settings.EntityID)
+	if err != nil {
+		return "", ctxerr.Wrap(ctx, err, "InitiateSSO creating authorization")
+	}
+
+	return idpURL, nil
+
+}
+
+func (svc *Service) InitiateMDMAppleSSOCallback(ctx context.Context, auth fleet.Auth) ([]byte, error) {
+	// skipauth: User context does not yet exist. Unauthenticated users may
+	// hit the SSO callback.
+	svc.authz.SkipAuthorization(ctx)
+
+	logging.WithLevel(logging.WithNoUser(ctx), level.Info)
+
+	appConfig, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "get config for sso")
+	}
+
+	_, metadata, err := svc.ssoSessionStore.Fullfill(auth.RequestID())
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "validate request in session")
+	}
+
+	err = sso.ValidateAudiences(
+		*metadata,
+		auth,
+		appConfig.SSOSettings.EntityID,
+		appConfig.ServerSettings.ServerURL,
+		appConfig.ServerSettings.ServerURL+svc.config.Server.URLPrefix+"/api/v1/fleet/mdm/sso/callback",
+	)
+
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "validating sso response")
+	}
+
+	return apple_mdm.GenerateEnrollmentProfileMobileconfig(
+		appConfig.OrgInfo.OrgName,
+		appConfig.ServerSettings.ServerURL,
+		svc.config.MDM.AppleSCEPChallenge,
+		svc.mdmPushCertTopic,
+	)
+}
+
+func (svc *Service) MDMAppleSyncDEPPRofile(ctx context.Context) error {
+	profiles, err := svc.ds.ListMDMAppleEnrollmentProfiles(ctx)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "listing profiles")
+	}
+
+	// Grab the first automatic enrollment profile we find, the current
+	// behavior is that the last enrollment profile that was uploaded is
+	// the one assigned to newly enrolled devices.
+	//
+	// TODO: this will change after #10995 where there can be a DEP profile
+	// per team.
+	var depProf *fleet.MDMAppleEnrollmentProfile
+	for _, prof := range profiles {
+		if prof.Type == "automatic" {
+			depProf = prof
+			break
+		}
```

Comment:
@mna heads-up that this conflicts with the work you're doing in #10995, apologies about that.

Reply:
:pray: thanks, no worries, chances are this will land before and I can rebase on top of this.